Add note to channel spec regarding channel subscription spec (#3132)
knative/eventing: docs/spec/channel.md

Each channel CRD MUST contain an array of subscribers:
[`spec.subscribers`](https://github.com/knative/eventing/blob/master/pkg/apis/duck/v1beta1/subscribable_types.go)

> Note: The array of subscribers MUST NOT be set directly on the generic
> Channel CR, but rather appended to the backing channel by the
> subscription itself.

Reviewer: <!--mdfmt-bot-->
Format markdown:
```suggestion
Note: The array of subscribers MUST NOT be set directly on the generic Channel
CR, but rather appended to the backing channel by the subscription itself.
```

Each channel CRD MUST contain an array of subscribers:
[`spec.subscribers`](https://github.com/knative/eventing/blob/master/pkg/apis/duck/v1beta1/subscribable_types.go)

Note: the array of subscribers MUST NOT be set directly on the channel, but

> rather appended by the subscription itself.

Reviewer 2: This means, we need to disable direct setting on IMC and other channels in the code as well.

Author: @aliok -- ok I've updated, please let me know what you think!

> CR, but rather appended to the backing channel by the subscription itself.

Reviewer 3: To match the rest of the doc:

```suggestion
custom object, but rather appended to the backing channel by the subscription itself.
```

> custom object, but rather appended to the backing channel by the subscription itself.

Reviewer: <!--mdfmt-bot-->
Format markdown:
```suggestion
custom object, but rather appended to the backing channel by the subscription
itself.
```

> custom object, but rather appended to the backing channel by the subscription itself.

Reviewer: <!--mdfmt-bot-->
Format markdown:
```suggestion
custom object, but rather appended to the backing channel by the subscription
itself.
```
